### PR TITLE
[Javascript SDK] Refactoring tests to check the Paths of the Mock servers

### PR DESCRIPTION
**Pull request by tonilopezmr**
### :tophat: What is the goal?
* Javascript SDK Tests refactor to check Paths and bodies on the mock server.
* Thanks for the new tests, I found a bug on the `Compute` create endpoint

### :memo: How is it being implemented?
* Abstract the `jest` mocks into a single object to use it in all tests, adding `expect` methods to Paths and Bodies.
* Remove unused code like `mockIdentify` which is already done in `mockServer`
* Remove duplicated code and moved to `mockServer`

### Usage
Now to mock any call you just have to call `mockServer` with the expected URL, and add the paths of every call with the return body or input body as well for POST.

BEFORE:

```javascript
test('get a key-value pair', async () => {
    seaFetch.mockImplementation((token: string) => ({
      ...authPost,
      get: (url: string) => {
        
        return textBody({
          "key":"Zm9vL2Jhcg",
          "value":"dmFsdWU"
        })
      }
    })) 
    
    expect(await metadata.get({key: "foo/bar"})).toStrictEqual({
      key: "foo/bar",
      value: "value"
    })    
 })

```

AFTER:

```javascript
test('get a key-value pair', async () => {
    server.get("/config/base64:Zm9vL2Jhcg", {
      "key":"Zm9vL2Jhcg",
      "value":"dmFsdWU"
    })

    expect(await metadata.get({key: "foo/bar"})).toStrictEqual({
      key: "foo/bar",
      value: "value"
    })    
 })
```

### :boom: How can it be tested?
* You can run the linter and tests using `nox` command
* Or run only tests by using `nox -s test`

#### Draft TODO list

[X] Check API call paths on every mock server call.
[X] Check Identify paths
[X] Check Metadata path API calls
[X] Check Locks path API calls
[X] Check Restrictions path API calls

### CODE REVIEW

**Review by Jess3Jane — APPROVED** [review]
I am largely going to trust it works as you say, it certainly makes things cleaner to my untrained eye!

File: seaplane-sdk/javascript/__tests__/api/helper.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    return textBody({token: "test_token"})
+  }
+}
+
+export const mockIdentify = (configuration: Configuration) => {  
+  seaFetch.mockImplementation((token: string) => ({
+    post: (url: string, body: string) => {                
+      return Promise.resolve({ 
+        ok: () => true,
+        text: () => Promise.resolve({token: "test_token"})
```
Comment (WesleyDRobinson) [inline seaplane-sdk/javascript/__tests__/api/helper.ts:27]: Help me understand why text() returns an object body with `{token: ""}`

**Review by WesleyDRobinson — DISMISSED** [review]
blocking for some questions and requested changes.

I may add some more comments, but want to issue this request for changes asap

I will also see about making a PR into this PR with some suggestions.

**Comment by WesleyDRobinson** [comment]
There is quite a bit of noise in my IDE due to questionable types, unused code, and disallowed `any` types; some other configuration issues also. So I have concerns about how this will actually execute, given that I'm not totally confident about the mocks.

For example: using `any` seems ok in a pinch but please make it explicit by allowing tsconfig.json or by surpressing with comments like `// ts-ignore <rule> for the reason`.

Ok, that being said, I don't want a bunch of existing errors and needed improvements to block progress on required work, so I'll remove my request for changes at this time.

File: seaplane-sdk/javascript/__tests__/api/restriction.spec.ts [hunk 2]
```diff
@@ -6,56 +6,36 @@ import { RestrictionState, SeaplaneApi } from '../../src/model/restrictions';
...
 
 describe('Given Restrictions API', () => {
 
   const config = new Configuration({ 
     apiKey: "test_apikey"
   })
   const restrictions = new Restrictions(config)  
-
-  beforeAll(() => {
-    mockIdentify(config)
```
Comment (WesleyDRobinson) [inline seaplane-sdk/javascript/__tests__/api/restriction.spec.ts:33]: this isn't replaced; is that intentional?

File: seaplane-sdk/javascript/src/api/compute.ts [hunk 3]
```diff
@@ -13,9 +13,12 @@ export default class Compute {
   }
 
   async create(formation: Formation): Promise<void> {
+    const gateway = formation.gateway_flight;
+    delete formation['gateway_flight'];
+
     const payload = {
       ...formation,
-      'gateway-flight': formation.gateway_flight,
```
Comment (WesleyDRobinson) [inline seaplane-sdk/javascript/src/api/compute.ts:18]: why change this?

i prefer the original, and i'd even consider that an unwanted mutation of `formation`
why `delete formation['gateway_flight']` ?

**Comment by WesleyDRobinson** [comment]
💥 How can it be tested?

    You can run the linter and tests using `npm run lint` command
    Or run only tests by using `npm run tests`

**Comment by tonilopezmr** [comment]
Oh sorry, I forgot to say you have to run `poetry shell` before !!

**Pull request merged**